client/gutenberg/extensions/presets/jetpack/utils/refresh-registrations.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { forEach, get, has } from 'lodash';
+import { getBlockType, registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import { getPlugin, registerPlugin, unregisterPlugin } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import getJetpackData from './get-jetpack-data';
+import extensions from '../editor';
+
+/**
+ * Refreshes registration of Gutenberg extensions (blocks and plugins)
+ *
+ * Uses block and plugin availability information obtained from the server to conditionally
+ * register and/or unregister blocks and plugins accordingly
+ *
+ * @returns {void}
+ */
+export default function refreshRegistrations() {
+	const extensionAvailability = get( getJetpackData(), [ 'available_blocks' ] );
+
+	if ( ! extensionAvailability ) {
+		return;
+	}
+
+	forEach( extensions, ( settings, name ) => {
+		const available = get( extensionAvailability, [ name, 'available' ] );
+
+		if ( has( settings, [ 'render' ] ) ) {
+			// If the extension has a `render` method, it's not a block but a plugin
+			const pluginName = `jetpack-${ name }`;
+			const registered = getPlugin( pluginName );
+
+			if ( available && ! registered ) {
+				registerPlugin( pluginName );
```

Comment:
Why not use the jetpackRegisterBlock function instead?

Reply:
Hmm, fair point. I think I'd need to change some of `jetpackRegisterBlock`'s internal logic around :thinking: 

TBH at this point I'd rather not touch the code much more for the sake of this PR since it took me a while to get some of the logic right :grimacing: Happy to revisit in a subsequent PR tho...